paper.md

```diff
@@ -2,11 +2,88 @@
+Ideally, workflows should improve the reproducibility of scientific applications 
+by making it easier to share and reuse between scientists. However, scientists 
+often find it difficult to reuse others workflows, which is known as _workflow decay_. [@workflow_decay]
+This is mostly because of the differences in the environment where the workflows are developed and the environment where they are being reproduced [@meng2017facilitating]. Some people have tried to solve this problem using the traditional Virtual machine model where they execute every task inside a separate VM [@gruning2018practical][@howe2012virtual]. People built virtual machine images from their development environment and used for setting up VM's in the Cloud. [@virtandnestedvirt2012]
+
+<!-- this is why it is still hard using VM's -->
+
+Although, the above approach solves the platform dependency problem very well, spawing 
+VM's to do granular steps of a workflow is very inefficient [@barik2016performance]. When several VM's are running on the same host, performance may be hindered if the computer it’s running on lacks sufficient power. This makes running workflows on commodity hardware extremely unlikely and inefficient. Resource utilization is very high in case of VM's 
+since every VM requires a copy of the operating system running on top of
+the host's operating system. In fact, a VM runs in a non-priviledged mode
+which does not have the capability to execute many privileged instructions. 
+Therefore, a hypervisor is needed to translate a VM instruction into an instruction 
+that can be executed by the host [@zhang2018comparative]. All these makes using VM's less scalable. Therefore all of these leads us to a question "Can we do better ?".
+
+<!-- what is container tech -->
+
+Although Software (Linux) containers are a relatively old technology [@menage_adding_2007], it was not until recently, with the rise of Docker, that they entered mainstream territory [@bernstein_containers_2014]. Since then, this technology has transformed the way applications get deployed in shared infrastructures, with 25% of companies using this form of software deployment [@datadog_surprising_2018], and a market size projected to be close to 5B by 2023 [@marketsandmarkets_application_2018]. Docker has been the de facto container runtime, with other container runtimes such as Singularity [@kurtzer_singularity_2017], Charliecloud [@priedhorsky_charliecloud_2017] and Podman [@podmancommunity_containers_2019] having emerged. The Linux Foundation bootstrapped the Open Container Initiative (OCI) [@opencontainerinitiative_new_2016] and is close to releasing version 1.0 of a container image and runtime specifications. With Docker, began the era of container-native software development, which is a paradigm that promotes the building, testing and deployment of softwares in containers. Software containers are well-known for being lightweight and
```

Comment:
some of these reads like copy-pasted. can you point where did you copy this from?

Reply:
@ivotron  I copied it from the introduction section of the original version of this paper. https://github.com/ivotron/popper2-paper/blob/e99c94bb4e06620c3e85c1159c0e20e450f9f251/paper.md#introduction-secintro